### PR TITLE
step 4,5

### PR DESCRIPTION
**Pull request by 0xAlison**
although I just completed two steps, the file has changed a lot(since the code format google gave us is  different from Intellij just like weiwei had mentioned). to avoid causing extra trouble for u(though I think still very troublesome:( , I'll explain every file I modify in detail below

### CODE REVIEW

**Comment by 0xAlison** [comment]
1. src/main/java/com/google/codeu/data/Datastore.java
basically add some user function in Datastore.java through the instruction on website AboutMe / step5: add datastore logic. 
image
2. src/main/java/com/google/codeu/data/User.java
add a new java file under dir: data. also from step 5. it's about User object
image

**Comment by 0xAlison** [comment]
3. src/main/java/com/google/codeu/servlets/AboutMeServlet.java
this is a newly-added file, but don't really need to check what has been deleted or added since the deleted codes are just for testing(learning) what Servlet is about(I think everyone has done the same practice as well). this is also the same code from step5.
image
4. src/main/webapp/aboutus.html
so this one is from step 4, I added the code to the wrong file, so I just deleted it.

File: src/main/webapp/js/user-page-loader.js [hunk 1]
```diff
@@ -73,26 +75,42 @@ function fetchMessages() {
...
 
 /** Fetches data and populates the UI of the page. */
 function buildUI() {
-  setPageTitle();
-  showMessageFormIfViewingSelf();
-  fetchMessages();
+    setPageTitle();
+    showMessageFormIfViewingSelf();
+    fetchMessages();
+    fetchAboutMe();
```
Comment (0xAlison) [inline src/main/webapp/js/user-page-loader.js:100]: added fetchAboutMe here

File: src/main/webapp/js/user-page-loader.js [hunk 2]
```diff
@@ -73,26 +75,42 @@ function fetchMessages() {
...
+        return response.text();
+    }).then((aboutMe) => {
+        const aboutMeContainer = document.getElementById('about-me-container');
+        if (aboutMe == '') {
+            aboutMe = 'This user has not entered any information yet.';
+        }
+
+        aboutMeContainer.innerHTML = aboutMe;
+
+    });
```
Comment (0xAlison) [inline src/main/webapp/js/user-page-loader.js:115]: here I added a new function fetchAboutMe(). according to step4

File: src/main/webapp/user-page.html [hunk 3]
```diff
@@ -16,32 +13,43 @@
...
-      <textarea name="text" id="message-input"></textarea>
-      <br/>
-      <input type="submit" value="Submit">
+<!--for aboutme box-->
+<div id="about-me-container">Loading...</div>
+<div id="about-me-form" class="hidden">
+    <form action="/about" method="POST">
+        <textarea name="about-me" placeholder="about me" rows=4 required></textarea>
+        <br/>
+        <input type="submit" value="Submit">
```
Comment (0xAlison) [inline src/main/webapp/user-page.html:38]: add a new message box according to step4
image
and this is how it works
image

**Review by 0xAlison — COMMENTED** [review]
for file 5 and 6, I have directly add some explanation onto the parts I have modified(the rest are just changed by the Intellij reformat)

**Comment by 0xAlison** [comment]
so overall this is how it works 
login first and then go to the "Your Page" then you'll see this page
image

**Review by Junochiu — APPROVED** [review]
Hi Alison, thanks for the detailed explanation! I found there’s a “hide white space change” option in git hub, may be that will help a little bit for reviewing the code on github :)

**Comment by nicolercl** [comment]
Hi Alison, I think you pushed two more commits before the pull request could be merged lol. The ones before those two are fine, but you might have missed the imports for cleaning user data, since it compiles with the error below:
螢幕快照 2019-06-02 下午1 59 00
Adding these lines in your servlet will make it work:
`import org.jsoup.Jsoup;`
`import org.jsoup.safety.Whitelist;`

**Comment by 0xAlison** [comment]
ahh yes! thanks a lot!I modify a line after Juno had checked but didn't noticed that! 
please check again! lots of thanks:)!

**Comment by nicolercl** [comment]
I think it's ready to merge~

**Review by nicolercl — APPROVED** [review]

**Pull request merged**